src/modm/platform/clock/rp/module.lb

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2016-2019, Niklas Hauser
+# Copyright (c) 2017, Fabian Greif
+# Copyright (c) 2021, Christopher Durand
+#
+# This file is part of the modm project.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+# -----------------------------------------------------------------------------
+
+
+def init(module):
+    module.name = ":platform:clockgen"
+    module.description = "Clock Generator and Source"
+
+def prepare(module, options):
+    if options[":target"].identifier.platform != "rp":
```

Comment:
```suggestion
    if options[":target"].has_driver("clocks:rp")
```
Allows you to switch on the peripheral type for free and it's shorter.
("clocks:rp20" with the suggested changes in modm-devices)